### PR TITLE
Refactor a few things in String+SourceKitten

### PR DESCRIPTION
**Pull request by jpsim**
This may have some performance implications. @norio-nomura what do you think? I pulled this out of #223.

### CODE REVIEW

**Comment by jpsim** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Error">
          1 Error
      </th>
     </tr>
  </thead>
  <tbody>

    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="true">Please include a CHANGELOG entry. <br />
You can find it at <a href="https://github.com/jpsim/SourceKitten/blob/master/CHANGELOG.md">CHANGELOG.md</a>.</td>
    </tr>
  </tbody>
</table>
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>

    <tr>
      <td>:warning:</td>
      <td data-sticky="true">This PR may need tests.</td>
    </tr>
  </tbody>
</table>
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>

    <tr>
      <td>:book:</td>
      <td data-sticky="true">Note, we hard-wrap at 80 chars and use 2 spaces after the last line.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">danger</a>
</p>

**Comment by jpsim** [comment]
After some basic unscientific testing, I think this slightly improves performance.

File: Source/SourceKittenFramework/String+SourceKitten.swift [hunk 1]
```diff
@@ -61,42 +61,32 @@ extension NSString {
...
+                let index = index + 1
+                let rangeStart = utf16CountSoFar
+                let utf16Count = content.utf16.count
+                utf16CountSoFar += utf16Count
+
+                let byteRangeStart = bytesSoFar
+                let byteCount = content.lengthOfBytesUsingEncoding(NSUTF8StringEncoding)
+                bytesSoFar += byteCount
+
+                let newlineLength = index != lineContents.count ? 1 : 0 // FIXME: assumes \n
```
Comment (norio-nomura) [inline Source/SourceKittenFramework/String+SourceKitten.swift]: Doesn't this work with DOS newline anymore?

Comment (jpsim) [inline Source/SourceKittenFramework/String+SourceKitten.swift]: that's right

**Pull request merged**